src/xopen/__init__.py

```diff
@@ -240,7 +243,12 @@ def __init__(  # noqa: C901
         else:
             if compresslevel is not None:
                 self._program_args += ["-" + str(compresslevel)]
-            self.outfile = open(path, mode[0] + "b")
```

Comment:
The `w` mode case is handled appropriately. The `r` case cannot accept filehandles, as the external program needs to open files. Please make sure there are test cases for multiple threads for pbzip2 and xz (which will defer to an external program) and ensure these cases are handled properly. In this case the PipedCompressionProgram implementation should raise an error that is appropriately handled in the calling function.

Reply:
I have some ideas  on how to make it work, but it will do it after your work is done :-).

Reply:
I've added a raise for now, as suggested.